src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -362,7 +362,14 @@ internal void SetBufferSizes(int readBufferSize, int writeBufferSize)
         public override void Flush()
         {
             if (_handle == null) InternalResources.FileNotOpen();
-            Interop.Termios.TermiosDiscard(_handle, Interop.Termios.Queue.AllQueues);
+
+            SpinWait sw = new SpinWait();
+            while (!(_readQueue.IsEmpty && _writeQueue.IsEmpty))
```

Comment:
why do we need to wait for internal queues to empty?

Reply:
So that scenario like:

```csharp
stream.Write(...);
stream.Flush(...);
```

will enforce that data was received - otherwise the Flush might finish before the write operation even gets started

Reply:
I actually have mixed feelings now if I should be doing this or not so let's talk offline

Reply:
Part I found confusing was readQueue.IsEmpty.Since that is flush on Stream, that would finish writing to layer bellow, right? 
Also once in write buffer, is there anything preventing data from being written?

Reply:
Is there a chance that calling an `IsEmpty` method gets stuck? In that case, how do we get out of the while loop?

Reply:
@carlossanlop this can theoretically only happen in multi-threaded scenario which we never guaranteed will work correctly but even in that case this would be a stress scenario which is very unlikely